### PR TITLE
CLOUDP-237968: Add authentication to MongoD

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
Added authentication to replica set.

Split `assets/run.sh` into 2 pieces:
- initialize
- running

Once initialized the initialization will not run again.
During the initialization we do not expose any ports to the outside.

### CODE REVIEW

File: assets/run.sh [hunk 1]
```diff
@@ -10,6 +9,19 @@ trap_forward_sig() {
 }
 
 MONGOD_HOST="localhost:27017"
+MONGOD_DB_PATH="/data/db"
+MONGOD_KEYFILE_PATH="/data/configdb/keyfile"
+MONGOD_PIDFILE_PATH="/var/run/mongod.pid"
+
+if [[ -z "$MONGODB_INITDB_ROOT_USERNAME" ]]; then
+    echo "Required environment variable MONGODB_INITDB_ROOT_USERNAME is not set"
```
Comment (fmenezes) [inline assets/run.sh:17]: authentication should be optional like our current image

File: assets/run.sh [hunk 2]
```diff
@@ -22,34 +34,80 @@ function kill_pids() {
...
+    INIT_PID=$!
+    PIDS+=("$INIT_PID")
+
+    wait_for_mongod
+
+    # Initialize the replica set
     mongosh "mongodb://$MONGOD_HOST" "/usr/local/etc/rs-init.js" > /dev/null 2>&1
+
+    # Insert the root user
+    mongosh "mongodb://$MONGOD_HOST/admin" --quiet --eval "db.createUser({user: '$MONGODB_INITDB_ROOT_USERNAME', pwd: '$MONGODB_INITDB_ROOT_PASSWORD', roles: [{role: 'root', db: 'admin'}]})" > /dev/null 2>&1
```
Comment (fmenezes) [inline assets/run.sh:101]: why not move this into the .js file? we have access to `process.env` for env vars if needed

File: assets/run.sh (hunk 1 above)
Comment (jeroenvervaeke) [reply assets/run.sh]: I was under the impression that authentication had to be enabled, but it was some issue in my bash script. Fixed it.

File: assets/run.sh (hunk 2 above)
Comment (jeroenvervaeke) [reply assets/run.sh]: Because we only want to insert users when authentication is enabled. See updated version.

Comment (fmenezes) [reply assets/run.sh]: we can have ifs in JavaScript

File: assets/run.sh [hunk 3]
```diff
@@ -22,34 +27,96 @@ function kill_pids() {
...
-
     trap_forward_sig kill_pids HUP TERM INT ABRT
+    
+    initialize
+
+    MONGOD_ARGS=("${MONGOD_BASE_ARGS[@]}")
+    if $AUTH_ENABLED; then
+        MONGOD_ARGS+=("--keyFile=$MONGOD_KEYFILE_PATH" "--auth")
+    else
+        MONGOD_ARGS+=("--noauth")
```
[the remark this answers is not in the document]
Comment (jeroenvervaeke) [reply assets/run.sh]: fixed

**Review by fmenezes — APPROVED** [review]

**Pull request merged**